File: dotnet/Solid/Solid/ISP/Example1/JsonStore.cs
using System;

namespace Solid.ISP.Example1
{
    public class JsonStore : IPersistenceMedium
    {
        public string FileName { get; set; }

        public string ConnectionString
        {
            get
            {
                throw new NotSupportedException("Not supported");
            }
            set
            {
                throw new NotSupportedException("Not supported");                
            }
        }

        public void Open()
        {
            Console.WriteLine("Opening file");
        }

        public string ExecuteQuery(string query)
        {
            throw new NotSupportedException("Not supported");
        }

        public string ReadFile()
        {
            return "File content!";
        }

        public void Close()
        {
            Console.WriteLine("Closing file");
        }
    }
}

File: java/isp-refacor/src/main/java/no/bouvet/solid/isp/JsonStore.java
package no.bouvet.solid.isp;

public class JsonStore implements FileConnection, FileOperation {

    private String fileName;

    @Override
    public String getFileName() {
        return fileName;
    }

    @Override
    public void setFileName(String fileName) {
        this.fileName = fileName;
    }

    @Override
    public void open() {
        // open file
    }

    @Override
    public String readFile() {
        return "File content!";
    }

    @Override
    public void close() {
        // close file
    }

}
